contrib/endpoints/src/api_manager/config_manager.cc

```diff
@@ -59,86 +51,69 @@ ConfigManager::ConfigManager(
-  if (!status.ok()) {
-    // We should not get here
-    global_context_->env()->LogError("Unexpected status: " + status.ToString());
-    return;
-  }
+void ConfigManager::FetchRollouts() {
+  service_management_fetch_->GetRollouts([this](const utils::Status& status,
+                                                std::string&& rollouts) {
+    if (!status.ok()) {
+      global_context_->env()->LogError(
+          std::string("Failed to download rollouts: ") + status.ToString() +
+          ", Response body: " + rollouts);
+      return;
+    }
 
-  rollouts_refresh_timer_ = global_context_->env()->StartPeriodicTimer(
-      std::chrono::milliseconds(refresh_interval_ms_), [this]() {
-        service_management_fetch_->GetRollouts([this](
-            const utils::Status& status, std::string&& rollouts) {
-          if (!status.ok()) {
-            global_context_->env()->LogError(
-                std::string("Failed to download rollouts: ") +
-                status.ToString() + ", Response body: " + rollouts);
-            return;
-          }
-
-          ListServiceRolloutsResponse response;
-          if (!utils::JsonToProto(rollouts,
-                                  (::google::protobuf::Message*)&response)
-                   .ok()) {
-            global_context_->env()->LogError(std::string("Invalid response: ") +
-                                             status.ToString() +
-                                             ", Response body: " + rollouts);
-            return;
-          }
-
-          if (response.rollouts_size() == 0) {
-            global_context_->env()->LogError("No active rollouts");
-            return;
-          }
-
-          std::shared_ptr<ConfigsFetchInfo> config_fetch_info =
-              std::make_shared<ConfigsFetchInfo>();
-
-          for (auto percentage :
-               response.rollouts(0).traffic_percent_strategy().percentages()) {
-            config_fetch_info->rollouts.push_back(
-                {percentage.first, round(percentage.second)});
-          }
-
-          if (config_fetch_info->rollouts.size() == 0) {
-            global_context_->env()->LogError("No active rollouts");
-            return;
-          }
-
-          FetchConfigs(config_fetch_info);
-        });
-      });
+    ListServiceRolloutsResponse response;
+    if (!utils::JsonToProto(rollouts, (::google::protobuf::Message*)&response)
+             .ok()) {
+      global_context_->env()->LogError(std::string("Invalid response: ") +
+                                       status.ToString() + ", Response body: " +
+                                       rollouts);
+      return;
+    }
+
+    if (response.rollouts_size() == 0) {
+      global_context_->env()->LogError("No active rollouts");
+      return;
+    }
+
+    if (previous_rollotus_id_ == response.rollouts(0).rollout_id()) {
+      return;
+    }
+    previous_rollotus_id_ = response.rollouts(0).rollout_id();
```

Comment:
I will like to call it "current_rollout_id",

start_esp needs to pass the rollout_id to api_manager too.   Can be in a separate PR.